mono/metadata/object.c

```diff
@@ -4510,12 +4510,14 @@ mono_object_new (MonoDomain *domain, MonoClass *klass)
  * For SGEN, these objects will only be freed at appdomain unload.
  */
 MonoObject *
-mono_object_new_pinned (MonoDomain *domain, MonoClass *klass)
+mono_object_new_pinned (MonoDomain *domain, MonoClass *klass, MonoError *error)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
 	MonoVTable *vtable;
 
+	mono_error_init (error);
+
 	vtable = mono_class_vtable (domain, klass);
 	if (!vtable)
 		return NULL;
```

Comment:
This is the same behavior as previously. The best fix for that case is to have and use `mono_class_vtable_checked`, but that's for another PR (https://github.com/mono/mono/pull/2455)